File: T4/codigos2122/Dialogos/app/src/main/java/com/example/dialogos/dialogos/DialogoComunicacion.java
package com.example.dialogos.dialogos;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

public class DialogoComunicacion extends DialogFragment {

    private String nombre;

    public static DialogoComunicacion newInstance(String nombre){

        DialogoComunicacion d = new DialogoComunicacion();
        Bundle bundle = new Bundle();
        bundle.putString("nombre",nombre);
        d.setArguments(bundle);
        // despues de retornar se ejecuta el ciclo de vida completo
        return d;
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        if (this.getArguments()!= null){
            this.nombre = this.getArguments().getString("nombre");
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
            // personalidazo
        // como puedo hacer para poner en el mensaje el nombre que me han pasado
        // por parametros
        builder.setMessage(nombre);
        return builder.create();
    }
}

File: Kotlin/T4/codigos_ret/Dialogos/app/src/main/java/com/example/dialogos/ui/dialog/DialogoComunicacion.kt
package com.example.dialogos.ui.dialog

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

class DialogoComunicacion : DialogFragment() {

    private lateinit var nombre: String;
    private lateinit var pass: String;

    companion object {
        fun newInstance(nombre: String, pass: String): DialogoComunicacion {
            val dialog: DialogoComunicacion = DialogoComunicacion();
            val bundle: Bundle = Bundle()
            bundle.putString("nombre", nombre)
            bundle.putString("pass", pass)
            dialog.arguments = bundle
            return dialog
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        // capturo los datos
        nombre = this.arguments?.getString("nombre") ?: "sin nombre"
        pass = this.arguments?.getString("pass") ?: "sin pass"
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder: AlertDialog.Builder = AlertDialog.Builder(requireActivity())
        builder.setTitle("Confirmacion")
        builder.setMessage("Por favor XXX, confirmas que tu pass es XXX")
        return builder.create()
    }

}